dev/TeachingTip/InteractionTests/TeachingTipTests.cs

```diff
@@ -0,0 +1,820 @@
+﻿using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Infra;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Common;
+using System;
+using System.Numerics;
+using Common;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+#if BUILD_WINDOWS
+using System.Windows.Automation;
+using MS.Internal.Mita.Foundation;
+using MS.Internal.Mita.Foundation.Controls;
+using MS.Internal.Mita.Foundation.Waiters;
+#else
+using Microsoft.Windows.Apps.Test.Automation;
+using Microsoft.Windows.Apps.Test.Foundation.Controls;
+using Microsoft.Windows.Apps.Test.Foundation.Waiters;
+#endif
+
+namespace Windows.UI.Xaml.Tests.MUXControls.InteractionTests
+{
+    [TestClass]
+    public class TeachingTipTests
+    {
+        // The longest observed animated view change took 5.4 seconds, so 9 seconds is picked
+        // as the default timeout so there is a reasonable margin for reliability.
+        const double defaultAnimatedViewChangeTimeout = 9000;
+
+        private struct TeachingTipTestPageElements
```

Comment:
I like this pattern a lot.

Even better would be to update the members to be properties, with a getter that finds the appropriate element and caches it.

Not a required change, just something to consider.

Reply:
Ooo yeah I like that much better.

---
In reply to: [252902602](https://github.com/Microsoft/microsoft-ui-xaml/pull/32#discussion_r252902602) [](ancestors = 252902602)